DoublyLinkedList/doubly_linked_list.h

```diff
@@ -0,0 +1,255 @@
+#include <iostream>
+using namespace std;
+// Node Creation
+class node
+{
+public:
+    int data;
+    node *next;
+    node *prev;
+};
+
+class doubly_linked_list : private node
+{ //class
+    node *head;
+    node *tail;
+    int length = 0;
+
+public:
+    doubly_linked_list()
+    {
+        head = NULL;
+    };
+    // insertion of value at front
+    void push_front(int value);
+    //insertion of value at end
+    void push_back(int value);
+
+    //for removing values at front and back
+    int remove_front();
+    int remove_back();
+
+    //for inserting at location
+    void insert(int value, int location);
+
+    //for removing at a location
+    void remove_at(int location);
+    void remove(int start, int end);
+
+    //for finding the size
+    size_t size();
+
+    //check whether empty or not
+    bool is_empty();
+
+    //Printing of values
+    void print_values();
+
+    //to clear all the elements in the list
+    void clear();
+
+    int printtail();
+};
+
+void doubly_linked_list::push_front(int value)
+{
+    //new node creation
+    node *new_node = new node();
```

Comment:
this and the next few lines are called together. can you add an appropriate contructor for node ?

Reply:
that avoids duplication ?